### PR TITLE
syntax: fix dircolors plugin to support termguicolors

### PR DESCRIPTION
**Pull request by msva**

### CODE REVIEW

File: runtime/syntax/dircolors.vim [hunk 1]
```diff
@@ -199,11 +199,11 @@ endfunction
 
 let b:dc_next_index = 0
 
-if has('gui_running')
+if (has('gui_running') || &termguicolors==1)
     call s:set_guicolors()
 endif
 
-if has('gui_running') || &t_Co != ''
+if (has('gui_running') || &termguicolors==1) || &t_Co != ''
```
Comment (justinmk) [inline runtime/syntax/dircolors.vim]: the parens are pointless and confusing here.

Comment (msva) [reply runtime/syntax/dircolors.vim]: Well, that was just a blind replace&check.
For now, I looked the file carefully and (kinda) fixed the issue.

What do you say about current state?

File: runtime/syntax/dircolors.vim [hunk 2]
```diff
@@ -84,7 +84,7 @@ endfunction
 
 function! s:get_hi_str(color, place) abort
     if a:color >= 0 && a:color <= 255
-        if has('gui_running')
+        if (has('gui_running') || &termguicolors==1)
```
Comment (justinmk) [inline runtime/syntax/dircolors.vim]: here too

Comment (msva) [reply runtime/syntax/dircolors.vim]: done

**Pull request merged**